[WIP] Add uaa curl command.

jmcarp opened the pull request:
I still need to flesh out the tests a bit if you think the approach is reasonable. WDYT @jhamon ?

jhamon commented:
This looks like a fantastic start. Other than missing test coverage, a couple things I noticed are

- The command won't be listed on the `uaa -h` help output unless you add a category for the command. This whole category thing is something I think cobra should support out of the box but it doesn't without hacking it in using Annotations and a custom usage template. Just put something like this in your command init method alongside the flags
```
	curlCmd.Annotations = make(map[string]string)
	curlCmd.Annotations[MISC_CATEGORY] = "true"
```

- You have code in place for zone switching, but because you didn't declare the --zone option on the command, I'm getting an error when I try to use it. More testing will catch errors like this one.

-  Seems like `--trace ` doesn't work even though it shows in `uaa curl -h`. Right now this is a global option showing in all help outputs because it's configured as a [persistent flag on the root command](https://github.com/cloudfoundry-incubator/uaa-cli/blob/ffb76f22e0abdaf20d9fd68c91d092a793ec13c3/cmd/root.go#L87). This was appropriate and useful until now because all the other crud stuff is relying, ultimately, on the same `doAndRead` method where the `config.Trace` property is read. (As an aside: I didn't know about `httputil.DumpResponse` and `httputil.DumpRequest` until reading your PR; I should definitely use those instead of reinventing the wheel on that inside doAndread.)

    I'm open to ideas on how we should handle this. I went with `trace` because that's what uaac called it, but it seems like maybe the trace flag should be renamed to `-v` and `--verbose` and implemented for this command so that it'd be consistent with real curl syntax. Thoughts?

jmcarp commented:
Thanks for taking a look @jhamon! I added more tests, dropped the zone logic (users can use `-H` to set zones), used `logger.Robots` for json, and borrowed some logic from `doAndRead` to support `--trace`. It might make more sense to refactor `doAndRead` for better reuse, but I didn't want to touch anything outside the `curl` module for now. Anyway, I'm happy to revise further if you have suggestions!

jhamon commented:
Thanks! This week has been kinda crazy, but I will try to look this over tomorrow and merge if nothing looks wonky.
